File: app/src/main/java/com/jbnm/homehero/MainActivity.java
package com.jbnm.homehero;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;

import com.jbnm.homehero.data.model.Child;
import com.jbnm.homehero.data.model.Parent;
import com.jbnm.homehero.data.model.Reward;
import com.jbnm.homehero.data.model.Task;
import com.jbnm.homehero.data.remote.DataManager;
import com.jbnm.homehero.ui.base.BaseActivity;
import com.jbnm.homehero.ui.goal.GoalActivity;
import com.jbnm.homehero.ui.taskpicker.TaskPickerActivity;

import java.util.Arrays;
import java.util.List;

import butterknife.ButterKnife;
import butterknife.OnClick;
import io.reactivex.Observer;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.disposables.Disposable;

public class MainActivity extends BaseActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);
    }

    @OnClick(R.id.buttonGoal)
    public void goalButton() {
        Intent intent = new Intent(MainActivity.this, GoalActivity.class);
        startActivity(intent);
    }

    @OnClick(R.id.buttonTask)
    public void taskButton() {
        startActivity(TaskPickerActivity.createIntent(this, Constants.CHILDID));
    }
}

File: app/src/main/java/com/jbnm/homehero/MainActivity.kt
package com.jbnm.homehero

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import com.jbnm.homehero.data.model.Task
import com.jbnm.homehero.ui.taskselector.TaskSelector

class MainActivity : AppCompatActivity() {
//    lateinit var taskWheel: TaskWheel
    lateinit var taskWheel: TaskSelector

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        taskWheel = findViewById(R.id.taskWheel)
        taskWheel.addTasks(generateTasks())
        taskWheel.setOnTaskSelectListener { task -> Toast.makeText(this, task.description, Toast.LENGTH_SHORT).show()}
    }

    fun generateTasks(): List<Task> {
        val instructions: List<String> = listOf("step1", "step2")
        val tasks: List<Task> = listOf(
                Task("testId", "Task 1", instructions, true),
                Task("testId", "Task 2", instructions, false),
                Task("testId", "Task 3", instructions, true),
                Task("testId", "Task 4", instructions, true),
                Task("testId", "Task 5", instructions, true),
                Task("testId", "Really long chore description", instructions, true),
                Task("testId", "Task 7", instructions, false)
        )
        return tasks
    }

}
